### DISCUSSION TITLE
Usecase Implementations

### DISCUSSION THREAD

**Discussion by arawho** [CSC207-2022F-UofT/course-project-justbob#47] [General]
There are a few usecases that remain to be implemented. Comment your usecases here:

Here is a set of pseudo instructions for how to implement a usecase. Make sure to check out the addAccount, archiveCourse and getAccountTrend usecases for examples on how these look. Try and keep your naming convention the same as the existing usecases.

1. Create a package for your usecase inside of the ports package (e.g., the archiveCourse package)
2. Create a request model for your usecase. This is a class with getters/setters for each piece of input data you need for the usecase.
3. Create a response model for your usecase. This is a class with getters/setters for each piece of output data you want to give back to the GUI. When deciding on what to include in the response model, think about what pieces of the GUI need to be updated now that your usecase has been performed.
4. Create the input boundary for your usecase. This is an interface with a single method called 'execute' that takes as input your request model, and returns as output a response model. It may also throw some errors if the usecase failed. Use custom errors as much as possible and mention them explicitly in the input boundary using the `throws` keyword.
5. Create the actual usecase in the usecases package. For this, have attributes describing the gateways/factories you need, and get that info in the constructor. implement the `execute` method, using any helpers you need to get the job done (make sure to check for errors in input!)
6. Add tests.

**Comment by samielsabri** [comment]
addCourseUseCase

**Comment by AlanC12138** [comment]
RemoveAccountUsecase

**Comment by samielsabri** [comment]
addAssessmentUseCase

**Reply by samielsabri** [reply]
SAMI

**Comment by samielsabri** [comment]
addAssessmentInstanceUseCase

**Reply by samielsabri** [reply]
SAMI

**Comment by samielsabri** [comment]
commitMarkUseCase

uncommitMarkUseCase

**Reply by samielsabri** [reply]
SAMI

**Comment by samielsabri** [comment]
submitInstanceUseCase

unsubmitInstanceUseCase

**Reply by samielsabri** [reply]
YUWEI

**Comment by samielsabri** [comment]
setMarkUseCase

**Reply by samielsabri** [reply]
SAMI

**Comment by DeepSJain** [comment]
getAccountTrendUseCase
getCourseTrendUseCase

**Comment by AlanC12138** [comment]
removeArchivedCourse

**Comment by samielsabri** [comment]
removeAssessmentUseCase

removeAssessmentInstanceUseCase

**Reply by vaishnav8791** [reply]
Vaishnav

**Comment by berrythebear** [comment]
writeToStickyNote
removeStickyNote
~~setNotification~~
~~removeNotification~~ (Was intended for eighth group member. Will do depending on time)

getDayData
getWeekData
getMonthData

toggleAssessments
toggleTimetable
toggleMiscellaneous

goToSeekedDate

**Reply by berrythebear** [reply]
Billy

**Comment by AlanC12138** [comment]
add and remove SemesterCourse

**Comment by samielsabri** [comment]
setInstanceMetaDataUseCase

**Reply by samielsabri** [reply]
SAMI

**Comment by samielsabri** [comment]
setCourseMetaDataUseCase

**Reply by samielsabri** [reply]
SAMI

**Comment by samielsabri** [comment]
setAssessmentTitleUseCase

**Reply by samielsabri** [reply]
SAMI

**Comment by arawho** [comment]
setAssessmentWeightScheme

**Reply by samielsabri** [reply]
SAMI
